Makefile

```diff
@@ -571,7 +574,18 @@ generate-metrics-config: $(ENVSUBST_BIN) ## Generate ./hack/observability/kube-s
 
 .PHONY: generate-diagrams
 generate-diagrams: ## Generate diagrams for *.plantuml files
```

Comment:
Not sure if it's a good idea, but wondering if we should run this in CI with a corresponding verify target
(I assume the resulting png files are reproducible)

Reply:
We're doing a basic verification - that the images don't disappear - using the link checker - I don't know if we really need something more advanced, but can look into it and take it as a follow up.

One issue is we also host a couple of svg images which are not generated by plantuml AFAIK

Reply:
I leave it up to you. I'm fine with just this PR

Reply:
We have a bunch of images that are not generated from plantuml - list is below:

I'm not going to add a verify target in this PR - but I'll open an issue mentioning the current issues. I think this was supposed to be followed up from https://github.com/kubernetes-sigs/cluster-api/pull/7033/ but was missed.

```
        deleted:    docs/book/src/images/describe-cluster-disable-grouping.png
        deleted:    docs/book/src/images/describe-cluster-echo.png
        deleted:    docs/book/src/images/describe-cluster-how-grouping-works.png
        deleted:    docs/book/src/images/describe-cluster-show-conditions.png
        deleted:    docs/book/src/images/describe-cluster.png
        deleted:    docs/book/src/images/introduction.png
        deleted:    docs/book/src/images/runtime-sdk-lifecycle-hooks.png
        deleted:    docs/proposals/images/capi-provider-operator/fig3.png
        deleted:    docs/proposals/images/capi-provider-operator/fig4.png
        deleted:    docs/proposals/images/clusterctl-redesign/components.png
        deleted:    docs/proposals/images/conditions/cluster-provision-workflow.png
        deleted:    docs/proposals/images/conditions/upgrade-workflow.png
        deleted:    docs/proposals/images/externally-managed-cluster-infrastructure/infrastructure.png
        deleted:    docs/proposals/images/in-place-propagation/current-state.png
        deleted:    docs/proposals/images/in-place-propagation/optional-changes.png
        deleted:    docs/proposals/images/in-place-propagation/proposed-changes.png
        deleted:    docs/proposals/images/ipam-integration/consumption.png
        deleted:    docs/proposals/images/ipam-integration/sequence.png
        deleted:    docs/proposals/images/machine-states-preboot/Figure2.png
        deleted:    docs/proposals/images/runtime-hooks/runtime-hooks.png
        deleted:    docs/proposals/images/runtime-sdk/overview.png
        deleted:    docs/proposals/images/runtime-sdk/swagger-ui.png
```

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/9331